Plugins/StomtPlugin/Source/StomtPlugin/Private/StomtAPI.cpp

```diff
@@ -315,139 +279,121 @@ FString UStomtAPI::ReadLogFile(FString LogFileName)
-					LoginRequestWasSend = false;
-				}
-				else
-				{
-					UE_LOG(LogTemp, Warning, TEXT("Login did not work"));
-				}
-			}
+			this->errorLog_file_uid = Request->GetResponseObject()->GetObjectField(TEXT("data"))->GetObjectField(TEXT("files"))->GetObjectField(TEXT("stomt"))->GetStringField("file_uid");
+			this->LogFileWasSend = false;
+			this->SendStomt(StomtToSend);
 		}
 	}
 
-	if (LogFileWasSend)
+	if (Request->GetResponseCode() == 403 || Request->GetResponseCode() == 419 || Request->GetResponseCode() == 413)
 	{
-		// Request access token
-		if (this->accesstoken.IsEmpty())
+		if (StomtToSend != NULL)
 		{
-			if (Request->GetResponseObject()->HasField(TEXT("meta")))
-			{
-				if (Request->GetResponseObject()->GetObjectField(TEXT("meta"))->HasField(TEXT("accesstoken")))
-				{
-					this->accesstoken = Request->GetResponseObject()->GetObjectField(TEXT("meta"))->GetStringField(TEXT("accesstoken"));
-					this->SaveAccesstoken(this->accesstoken);
-				}
-			}
+			this->SendStomt(StomtToSend);
+			LogFileWasSend = false;
 		}
+	}
+}
+
+void UStomtAPI::SendEMail(FString EMail)
+{
+	UStomtRestRequest* request = this->SetupNewPostRequest();
+	request->OnRequestComplete.AddDynamic(this, &UStomtAPI::UStomtAPI::OnSendEMailResponse);
+
+	request->GetRequestObject()->SetStringField(TEXT("email"), EMail);
 
-		// Get File uid of the error log and send stomt
+	request->ProcessURL(this->GetRestURL().Append(TEXT("/authentication/subscribe")));
+}
+
+
+void UStomtAPI::OnSendEMailResponse(UStomtRestRequest * Request)
+{
+	if (Request->GetResponseCode() != 400)
+	{
 		if (Request->GetResponseObject()->HasField(TEXT("data")))
 		{
-			if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->HasField(TEXT("files")))
+			if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->HasField(TEXT("success")))
 			{
-				this->errorLog_file_uid = Request->GetResponseObject()->GetObjectField(TEXT("data"))->GetObjectField(TEXT("files"))->GetObjectField(TEXT("stomt"))->GetStringField("file_uid");
-				this->LogFileWasSend = false;
-				this->SendStomt(StomtToSend);
+				if (Request->GetResponseObject()->GetObjectField(TEXT("data"))->GetBoolField("success"))
+				{
+					this->Config->SetSubscribed(true);
+				}
 			}
-		}
-
-		if (Request->GetResponseCode() == 403 || Request->GetResponseCode() == 419 || Request->GetResponseCode() == 413)
-		{
-			if (StomtToSend != NULL)
+			else
 			{
-				this->SendStomt(StomtToSend);
-				LogFileWasSend = false;
+				this->Config->SetSubscribed(false);
 			}
 		}
 	}
+	else
+	{
+		this->Config->SetSubscribed(true);
+	}
+}
+
+void UStomtAPI::SendLogoutRequest()
+{
+	UStomtRestRequest* request = this->SetupNewDeleteRequest();
+	request->OnRequestComplete.AddDynamic(this, &UStomtAPI::UStomtAPI::OnSendLogoutResponse);
 
-	if (EMailFlagWasSend)
+	request->ProcessURL(this->GetRestURL().Append(TEXT("/authentication/session")));
+}
+
+void UStomtAPI::OnSendLogoutResponse(UStomtRestRequest * Request)
```

Comment:
We don't have to wait that the logout request is successful, we can update the config at the same time we send the logout request.